x/leverage/types/msgs.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/umee-network/umee/v3/util/checkers"
+)
+
+var (
+	_ sdk.Msg = &MsgUpdateRegistry{}
+)
+
+// NewMsgUpdateRegistry will creates a new MsgUpdateRegistry instance
+func NewMsgUpdateRegistry(creator, title, description string, registry []Token) *MsgUpdateRegistry {
+	return &MsgUpdateRegistry{
+		Title:       title,
+		Description: description,
+		Registry:    registry,
+		Creator:     creator,
+	}
+}
+
+// Route implements Msg
+func (msg MsgUpdateRegistry) Route() string { return types.RouterKey }
+
+// Type implements Msg
+func (msg MsgUpdateRegistry) Type() string { return sdk.MsgTypeURL(&msg) }
+
+// ValidateBasic implements Msg
+func (msg MsgUpdateRegistry) ValidateBasic() error {
+	for _, token := range msg.Registry {
+		if err := token.Validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
For security, shouldn't this also verify that `msg.Creator` is equal to gov module address?

Reply:
while create the proposal , gov tx will check the GetSigners is the gov module account address

Reply:
There's no way for an attacker to submit this message directly to the rpc, even if they (for example) fork the CLI to be able to create the message?

Reply:
I will check this one